src/tr1/game/savegame/savegame_legacy.c

```diff
@@ -692,147 +578,13 @@ bool Savegame_Legacy_LoadOnlyResumeInfo(MYFILE *fp, GAME_INFO *game_info)
-        .num_leadbar = Inv_RequestItem(O_LEADBAR_ITEM),
-        0
-    };
-
-    M_Write(&item_stats, sizeof(item_stats));
-
-    int32_t flip_status = Room_GetFlipStatus();
-    M_Write(&flip_status, sizeof(int32_t));
-    for (int i = 0; i < MAX_FLIP_MAPS; i++) {
-        int8_t flag = Room_GetFlipSlotFlags(i) >> 8;
-        M_Write(&flag, sizeof(int8_t));
-    }
-
-    for (int32_t i = 0; i < Camera_GetFixedObjectCount(); i++) {
-        const OBJECT_VECTOR *const object = Camera_GetFixedObject(i);
-        M_Write(&object->flags, sizeof(int16_t));
-    }
-
-    Savegame_ProcessItemsBeforeSave();
-
-    for (int32_t i = 0; i < Item_GetLevelCount(); i++) {
-        ITEM *const item = Item_Get(i);
-        const OBJECT *const obj = Object_Get(item->object_id);
-
-        if (obj->save_position) {
-            M_Write(&item->pos.x, sizeof(int32_t));
-            M_Write(&item->pos.y, sizeof(int32_t));
-            M_Write(&item->pos.z, sizeof(int32_t));
-            M_Write(&item->rot.x, sizeof(int16_t));
-            M_Write(&item->rot.y, sizeof(int16_t));
-            M_Write(&item->rot.z, sizeof(int16_t));
-            M_Write(&item->room_num, sizeof(int16_t));
-            M_Write(&item->speed, sizeof(int16_t));
-            M_Write(&item->fall_speed, sizeof(int16_t));
-        }
-
-        if (M_ItemHasSaveAnim(item)) {
-            M_Write(&item->current_anim_state, sizeof(int16_t));
-            M_Write(&item->goal_anim_state, sizeof(int16_t));
-            M_Write(&item->required_anim_state, sizeof(int16_t));
-            M_Write(&item->anim_num, sizeof(int16_t));
-            M_Write(&item->frame_num, sizeof(int16_t));
-        }
-
-        if (M_ItemHasHitPoints(item)) {
-            M_Write(&item->hit_points, sizeof(int16_t));
-        }
-
-        if (M_ItemHasSaveFlags(obj, item)) {
-            uint16_t flags = item->flags + item->active + (item->status << 1)
-                + (item->gravity << 3) + (item->collidable << 4);
-            if (obj->intelligent && item->data) {
-                flags |= SAVE_CREATURE;
-            }
-            M_Write(&flags, sizeof(uint16_t));
-            M_Write(&item->timer, sizeof(int16_t));
-            if (flags & SAVE_CREATURE) {
-                CREATURE *creature = item->data;
-                M_Write(&creature->head_rotation, sizeof(int16_t));
-                M_Write(&creature->neck_rotation, sizeof(int16_t));
-                M_Write(&creature->maximum_turn, sizeof(int16_t));
-                M_Write(&creature->flags, sizeof(int16_t));
-                M_Write(&creature->mood, sizeof(int32_t));
-            }
-        }
-    }
-
-    M_WriteLara(&g_Lara);
-
-    int32_t flip_effect = Room_GetFlipEffect();
-    int32_t flip_timer = Room_GetFlipTimer();
-    M_Write(&flip_effect, sizeof(int32_t));
-    M_Write(&flip_timer, sizeof(int32_t));
-
-    File_WriteData(fp, buffer, m_SGBufPos);
-    Memory_FreePointer(&buffer);
+    Shell_ExitSystem("Legacy save writing is not supported.");
```

Comment:
We can just `ASSERT(false)` or leave the body as empty, or even not assign `Savegame_Legacy_SaveToFile` at all and leave it as `nullptr`.

Reply:
Thanks, went with `nullptr` and added an extra guard in `savegame.c` just in case.